tests/components/recorder/test_purge.py

```diff
@@ -306,27 +318,44 @@ async def test_purge_method(
         assert statistics_runs.count() == 7
         statistic_runs_before_purge = statistics_runs.all()
 
-        await hass.async_block_till_done()
-        await async_wait_purge_done(hass, instance)
+        for itm in runs_before_purge:
+            session.expunge(itm)
+        for itm in statistic_runs_before_purge:
+            session.expunge(itm)
 
-        # run purge method - no service data, use defaults
-        await hass.services.async_call("recorder", "purge")
-        await hass.async_block_till_done()
+    await hass.async_block_till_done()
+    await async_wait_purge_done(hass, instance)
 
-        # Small wait for recorder thread
-        await async_wait_purge_done(hass, instance)
+    # run purge method - no service data, use defaults
+    await hass.services.async_call("recorder", "purge")
+    await hass.async_block_till_done()
```

Comment:
We can't do this when holding the session since the pytest thread has an open connection, preventing connections from the recorder thread